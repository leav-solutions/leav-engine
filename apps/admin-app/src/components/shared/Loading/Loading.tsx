// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import React from 'react';
import {useTranslation} from 'react-i18next';
import {Dimmer, Loader, SemanticSIZES} from 'semantic-ui-react';

interface ILoadingProps {
    withDimmer?: boolean;
    size?: SemanticSIZES;
}

const Loading = ({withDimmer, size}: ILoadingProps): JSX.Element => {
    const {t} = useTranslation();
    const loader = (
        <Loader active inline="centered" size={size}>
            {t('admin.loading')}
        </Loader>
    );

    return withDimmer ? (
        <Dimmer active inverted>
            {loader}
        </Dimmer>
    ) : (
        loader
    );
};

export default Loading;
