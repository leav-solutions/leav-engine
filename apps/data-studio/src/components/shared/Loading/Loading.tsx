// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {Spin} from 'antd';
import styled from 'styled-components';

const LoadingElem = styled(Spin)`
    && {
        display: block;
        margin: 3em;
    }
`;

function Loading(): JSX.Element {
    return <LoadingElem data-testid="loading" />;
}

export default Loading;
