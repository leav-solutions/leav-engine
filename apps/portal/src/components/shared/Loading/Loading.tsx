// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {Spin} from 'antd';
import React from 'react';
import styled from 'styled-components';

const LoadingElem = styled(Spin)`
    && {
        display: block;
        margin: 3em;
    }
`;

function Loading(): JSX.Element {
    return <LoadingElem />;
}

export default Loading;
