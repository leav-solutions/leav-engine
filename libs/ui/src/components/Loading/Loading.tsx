// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {Spin} from 'antd';
import styled from 'styled-components';

interface ILoadingProps {
    compact?: boolean;
}

const LoadingElem = styled(Spin)<ILoadingProps>`
    && {
        display: block;
        margin: ${props => (props.compact ? '1em' : '3em')};
        font-size: inherit;
    }
`;

function Loading(props: ILoadingProps): JSX.Element {
    return <LoadingElem {...props} />;
}

export default Loading;
