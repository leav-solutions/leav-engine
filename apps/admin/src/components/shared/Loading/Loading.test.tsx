// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {shallow} from 'enzyme';
import React from 'react';
import {create} from 'react-test-renderer';
import Loading from '../../shared/Loading';

describe('Loading', () => {
    test('Snapshot test', async () => {
        const comp = create(<Loading />).toJSON();

        expect(comp).toMatchSnapshot();
    });

    test('Display dimmer', () => {
        const comp = shallow(<Loading withDimmer />);

        expect(comp.find('Dimmer').length).toBe(1);
    });
});
