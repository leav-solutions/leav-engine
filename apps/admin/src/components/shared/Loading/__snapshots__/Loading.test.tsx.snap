// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Snapshot test 1`] = `
<div
  className="ui active text centered inline loader"
>
  admin.loading
</div>
`;
